src/OpenDiffix.Core/Expression.fs

```diff
@@ -220,18 +220,22 @@ let rec evaluate (row: Row) (expr: Expression) =
 
 /// Sorts a row sequence based on the given orderings.
 let sortRows orderings (rows: Row seq) =
-  let rec compare orderings rowA rowB =
-    match orderings with
+  let exprsComparers =
+    orderings
+    |> List.map (fun (OrderBy (expr, direction, nulls)) -> expr, Value.comparer direction nulls)
+
+  let rec compare exprsComparers rowA rowB =
+    match exprsComparers with
     | [] -> 0
-    | OrderBy (expr, direction, nulls) :: tail ->
+    | (expr, comparer) :: tail ->
       let valueA = evaluate rowA expr
       let valueB = evaluate rowB expr
 
-      match Value.comparer direction nulls valueA valueB with
+      match comparer valueA valueB with
```

Comment:
Hm, this seems to be just a refactoring, not an optimization: the closure is still instantiated each time (and has to be), isn't it?

Reply:
`Value.comparer direction nulls` computes some constants before returning a comparer function.

`(Value.comparer direction nulls) -> returns func -> called with valueA valueB`